packages/shared/src/hooks/useJoinReferral.ts

```diff
@@ -0,0 +1,60 @@
+import request from 'graphql-request';
+import { useRouter } from 'next/router';
+import { useQuery } from 'react-query';
+import { expireCookie, setCookie } from '../lib/cookie';
+import { isDevelopment } from '../lib/constants';
+import { GET_REFERRING_USER_QUERY } from '../graphql/users';
+import { graphqlUrl } from '../lib/config';
+import { ApiErrorResult } from '../graphql/common';
+import { useAuthContext } from '../contexts/AuthContext';
+import { disabledRefetch } from '../lib/func';
+
+export const useJoinReferral = (): void => {
+  const { user, refetchBoot, isAuthReady } = useAuthContext();
+  const router = useRouter();
+  const { cid, userid } = router.query;
+  const shouldSetReferralCookie = isAuthReady && !user;
+
+  useQuery(
```

Comment:
Optional, but maybe open for discussion...

I personally find it more readable to have a function defined that handles the logic and the useQuery to simply invoke that function.
Would love to hear if we can generalise one way moving forward.